lib/datadog/core/telemetry/metric_queue.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+module Datadog
+  module Core
+    module Telemetry
+      # Stores all the metrics information by request_type during a time interval
+      class MetricQueue
+        attr_reader :metrics
+
+        def initialize
+          @metrics = {
+            'generate-metrics' => {},
+            'distributions' => {},
+          }
+        end
```

Comment:
Is this data structure a bit of a data race?

* Can multiple threads be writing to it through `@metrics` at the same time?
* Can the reader thread calling `build_metrics_payload` run at the same time as a metric is being written?

Reply:
Indeed, it is not thread-safe 😄 

Let me change it a bit to make it thread-safe 🔒

Reply:
Done https://github.com/DataDog/dd-trace-rb/pull/3230/commits/1574d153ef9ab931c46b1886eac54d3ef557a19e